File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var countNegatives = function(grid) {
  const m = grid.length
  const n = grid[0].length
  let ans = 0
  for (let i = 0; i < m; i++) {
    const index = grid[i].findIndex(el => el < 0)
    if (index >= 0) ans += n - index    
  }
  return ans
};

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.go
func findFirstNegativeIndex(row []int, l int) int {
  r := len(row) - 1
  for l <= r {
    mid := (l + r) / 2
    if row[mid] >= 0 {
      l = mid + 1
    } else {
      r = mid - 1
    }
  }
  return l
}
func countNegatives(grid [][]int) int {
  m := len(grid)
  n := len(grid[0])
  j := 0
  ans := 0
  for i := m - 1; i >= 0; i -- {
    j = findFirstNegativeIndex(grid[i], j)
    ans += n - j
  }
  return ans
}